doc/user-manual/language/telescopes.lagda.rst

```diff
@@ -2,14 +2,104 @@
   ::
   module language.telescopes where
 
+  open import Agda.Builtin.Bool
+  open import Agda.Builtin.Nat
+
+  postulate
+    List : Set → Set
+    _++_ : {A : Set} → List A → List A → List A
+    Vec : Set → Nat → Set
+    IsTrue : Bool → Set
+    Monoid : Set → Set
+    NonZero : Nat → Set
+
+  record _×_ (A B : Set) : Set where
+    constructor _,_
+    field
+      fst : A
+      snd : B
+
 .. _telescopes:
 
 **********
 Telescopes
 **********
 
+A telescope is a non-empty sequence of variable bindings annotated by their types,
+with each variable surrounded by parentheses.
+For example, ``(x : Nat) (y : Bool) (z : Bool)`` is a telescope.
+Adjacent variables that have the same type can share a type annotation.
+For example, the same telescope can be written equivalently as ``(x : Nat) (y z : Bool)``.
+The type of each variable can depend on the previous variables in the telescope,
+for example ``(A : Set) (n : Nat) (v : Vec A n)``.
+
 .. note::
-   This is a stub.
+  The terminology is due to de Bruijn :ref:`[1] <telescopes-refs>`:
+  "The word was inspired, of course, by the old-fashioned instrument consisting of segments that slide one into another."
+  Each variable binding corresponds to a segment of the telescope, which can slide into (i.e. depend on) the previous ones.
+
+Telescopes appear in the following parts of the Agda syntax:
+
+* :ref:`Function types<function-types>`
+* Declarations of :ref:`data types<data-types>` and :ref:`record types<record-types>`
+* Declarations of :ref:`parameterised modules <parameterised-modules>`
+
+::
+
+  postulate
+    f : (A : Set) (n : Nat) (v : Vec A n) → Nat
+
+  data D (A : Set) (n : Nat) (v : Vec A n) : Set where
+    -- ...
+
+  module M (A : Set) (n : Nat) (v : Vec A n) where
+    -- ...
+
+
+In telescopes of data and record types as well as parameterised modules,
+it is allowed to omit the type of a variable binding. This is equivalent
```

Comment:
It is also allowed in function types if you prefix the telescope with \forall.

Reply:
Yes, that's what it says on the line right below.

Reply:
Oh yes!